stubs/entrypoints/entrypoints.pyi

```diff
@@ -0,0 +1,58 @@
+import sys
+from typing import Any, Dict, Iterator, List, Sequence, Text, Tuple, Type, TypeVar
+
+if sys.version_info >= (3, 0):
+    from configparser import ConfigParser
+else:
+    from ConfigParser import ConfigParser
+
+if sys.version_info >= (3, 8):
+    from re import Pattern
+else:
+    from typing import Pattern
+
+_E = TypeVar("_E", bound="EntryPoint")
+
+entry_point_pattern: Pattern[Text]
+file_in_zip_pattern: Pattern[Text]
+
+class BadEntryPoint(Exception):
+    epstr: Text
+    def __init__(self, epstr: Text) -> None: ...
+    @staticmethod
+    def err_to_warnings() -> Iterator[None]: ...
+
+class NoSuchEntryPoint(Exception):
+    group: Text
+    name: Text
+    def __init__(self, group: Text, name: Text) -> None: ...
+
+class EntryPoint:
+    name: Text
+    module_name: Text
+    object_name: Text
+    extras: Sequence[Text] | None
+    distro: Distribution | None
+    def __init__(
+        self,
+        name: Text,
+        module_name: Text,
+        object_name: Text,
+        extras: Sequence[Text] | None = ...,
+        distro: Distribution | None = ...,
+    ) -> None: ...
+    def load(self) -> Any: ...
+    @classmethod
+    def from_string(cls: Type[_E], epstr: Text, name: Text, distro: Distribution | None = ...) -> _E: ...
```

Comment:
You can just use `_typeshed.Self`.

Reply:
@JelleZijlstra wow, this is awesome, thank you! Learned something new again, will check tomorrow whether I can use that in our company's type hints.

Reply:
It's not so great outside of typeshed, because you can't import `_typeshed` outside of `if TYPE_CHECKING`, and at runtime you may actually want a TypeVar with a bound. But it's really nice in stubs :)